File: tests/csharp/tests-csharp-xunit-api_request-playwright/Helpers/LayoutCss.cs
using System.Text.RegularExpressions;

namespace Helpers;

public static class LayoutCss
{
    /// <summary>Must match <c>@media (max-width: …)</c> in frontend/css/header.css.</summary>
    public const int ResponsiveBreakpointPx = 768;

    /// <summary>Desktop layout starts at breakpoint + 1px.</summary>
    public const int WideLayoutMinViewportPx = ResponsiveBreakpointPx + 1;

    private static readonly Regex RepeatColumns = new(@"repeat\((\d+)", RegexOptions.Compiled);

    public static int GridColumnCount(string? gridTemplateColumns)
    {
        if (string.IsNullOrWhiteSpace(gridTemplateColumns) || gridTemplateColumns.Trim() == "none")
        {
            return 0;
        }

        var normalized = gridTemplateColumns.Trim();
        var match = RepeatColumns.Match(normalized);
        if (match.Success)
        {
            return int.Parse(match.Groups[1].Value);
        }

        return normalized.Split((char[]?)null, StringSplitOptions.RemoveEmptyEntries).Length;
    }
}

File: tests/typescript/tests-typescript-api_request-playwright/src/helpers/layout-css.ts
export const RESPONSIVE_BREAKPOINT_PX = 768;

const REPEAT_COLUMNS = /repeat\((\d+)/;

export function gridColumnCount(gridTemplateColumns: string | null | undefined): number {
  if (gridTemplateColumns == null) {
    return 0;
  }
  const normalized = String(gridTemplateColumns).trim();
  if (!normalized || normalized === 'none') {
    return 0;
  }
  const match = normalized.match(REPEAT_COLUMNS);
  if (match) {
    return Number.parseInt(match[1], 10);
  }
  return normalized.split(/\s+/).length;
}
